apps/etterlatte-behandling/src/main/kotlin/config/BeanFactory.kt

```diff
@@ -269,4 +280,21 @@ class EnvBasedBeanFactory(private val env: Map<String, String>) : CommonFactory(
             minutterGamleHendelser = env.getValue("HENDELSE_MINUTTER_GAMLE_HENDELSER").toLong()
         ).schedule()
     }
+
+    override fun norg2HttpClient(): Norg2Klient {
+        return Norg2KlientImpl(httpClient(), env.getValue("NORG2_URL"))
+    }
+
+    override fun featureToggleService(): FeatureToggleService {
+        return FeatureToggleService.initialiser(
+            mapOf(
+                FeatureToggleServiceProperties.ENABLED.navn to config.getString("funksjonsbrytere.enabled"),
+                FeatureToggleServiceProperties.APPLICATIONNAME.navn to config.getString(
+                    "funksjonsbrytere.unleash.applicationName"
+                ),
+                FeatureToggleServiceProperties.URI.navn to config.getString("funksjonsbrytere.unleash.uri"),
```

Comment:
Trur ikkje det skal vera nødvendig å sende inn `APPLICATIONNAME` og `CLUSTER`. Og viss det er det bør vi sjå på initialiseringskoden igjen, for desse vil jo vera `NAIS_CLUSTER_NAME` og `NAIS_APP_NAME` i alle applikasjonane.

Reply:
Det fulgte jeg ikke helt med på.

cluster trengs vel for å matche på cluster strat.
name brukes for at vi ser riktig navn i unleash gui

Så lastes de via config - i application conf så er de satt til:

funksjonsbrytere.unleash.cluster = ${?NAIS_CLUSTER_NAME}
funksjonsbrytere.unleash.applicationName = ${?NAIS_APP_NAME}

som jeg tror leser dem fra env inn i config ( ${?...} syntaksen ) ?

Reply:
Var kanskje litt uklar, ser for meg noko sånt:
```
override fun featureToggleService(): FeatureToggleService {
        return FeatureToggleService.initialiser(
            mapOf(
                FeatureToggleServiceProperties.ENABLED.navn to config.getString("funksjonsbrytere.enabled"),                                 
                FeatureToggleServiceProperties.URI.navn to config.getString("funksjonsbrytere.unleash.uri")
            )
        )
```
Og at du i application.conf kun treng å leggje til
`funksjonsbrytere.enabled = "true"
funksjonsbrytere.unleash.uri = "https://unleash.nais.io/api/"`

Reply:
Men i initialiser har du:


```
cluster = env[FeatureToggleServiceProperties.CLUSTER.navn] ?: throw IllegalArgumentException("Unleash-cluster er ikke definert"),
applicationName = env[FeatureToggleServiceProperties.APPLICATIONNAME.navn] ?: throw IllegalArgumentException("Unleash-applikasjonsnavn er ikke definert")
```

?

Og - hvis ikke de settes - så vil ikke vi kunne bruke dem i hhv

appName(unleash.applicationName)

ByClusterStrategy(unleash.cluster)

Reply:
Tanken (usikker på om eg har fått til å initialisere det rett) er at dei skal få verdiane frå `NAIS_CLUSTER_NAME` 
 og `NAIS_APP_NAME` - som er to properties som allereie fins - automatisk.

Reply:
Den leser bare env: Map<String, String> - så hvor de skal komme fra vet jeg ikke helt

Reply:
Kan prøve meg på ein PR på det, har nokre idear til korleis vi kan løyse det.

Reply:
Gjerne.

Det jeg har gjort her er:

Flyttet init inn i companion object
Fikset as Boolean -> toBoolean (for å unngå en classcastexception jeg fikk)
Omdøpt Properties til en mer bestemt navn.

Reply:
Ref diskusjon - enig at det skal ryddes - men tas i egen PR litt senere.